script/urlChecker.ts

```diff
@@ -38,15 +44,23 @@ async function checkPages(bookSlug: string, pages: string[]) {
   const notFound: string[] = [];
 
   const visitPage = async(page: string) => {
-    try {
-      const response = await fetch(`${rootUrl}${page}`);
-      if (response.status === 404) {
+    const checkIfPageDoesntExist = async(url: string) => {
+      try {
+        const response = await fetch(url);
+        return response.status === 404;
+      } catch {
+        anyFailures = true;
+        bar.interrupt(`- (error loading) ${page}`);
+      }
+    };
+
+    if (checkIfPageDoesntExist(`${rootUrl}${page}`)) {
```

Comment:
I think there is no need to first go to the page and then go to the redirected page in case of 404
You should do the same as in `domVisitior` where you first check for the redirected url